docs/block-processing/overall.md

```diff
@@ -0,0 +1,105 @@
+# Block processing
+
+## Task definition
+
+Block processing can be described by presenting an algorithm to solve the following problem:
+
+> Given a sequence of blocks `B₀, B₁, B₂, …` (where `B₀` is the first
+genesis block) check whether block payload from these blocks are valid.
+
+We describe block processing as stateful algorithm:
+* Initial state `S₀` is derived from blockchain genesis data (see [mainnet genesis JSON](https://raw.githubusercontent.com/input-output-hk/cardano-sl/e7cfb1724024e0db2f25ddd2eb8f8f17c0bc497f/node/mainnet-genesis.json))
+* `S₁, S₂, …` are maintained as sequential application of blocks  `B₀, B₁, B₂, … ` to state `S₀`
+* State transition function. Given GState `S` and a main block `B` return either an error describing why `B` is invalid (w.r.t. tx payload) or new GState `S'`
+  ```
+  verifyAndApplyGState :: GState -> MainBlock -> Either BlockPayloadVerificationError GState
+  -- ^ Note, that function definition and types are different in code and are put here for reader's convenience
+  ```
+  * Note, that state transition function is defined only for main blocks. This is done because genesis blocks 
+  (as defined in Ouroboros) don't have payload.
+
+Note, that in theory payload verification can be stateless (except genesis
+state `S₀`) and be described without mentioning any additional state
+(apart from blocks themselves). But in practice it would be very
+inefficient and inconvenient.
+
+For sake of simplicity, we describe state transition function in two parts:
+* Verification: given GState `S` and a main block `B`, check whether `B` is valid (and can be applied to `S`)
```

Comment:
Not only main block. Even now (when we cover only txp and US) genesis blocks are important, because US has a lot of logic when processing genesis block.